sqlest/src/test/scala/sqlest/ast/BaseColumnTypeEquivalenceSpec.scala

```diff
@@ -0,0 +1,419 @@
+    ColumnTypeEquivalence.alignColumnTypes(bigDecimalColumn, Some(0.1).constant)._2.asInstanceOf[ConstantColumn[_]].value should be(0.1)
+    ColumnTypeEquivalence.alignColumnTypes(bigDecimalColumn, Some(BigDecimal("0.1")).constant)._2.asInstanceOf[ConstantColumn[_]].value should be(BigDecimal("0.1"))
+    ColumnTypeEquivalence.alignColumnTypes(mappedYNBooleanColumn, Some(false).constant)._2.asInstanceOf[ConstantColumn[Boolean]].value should be(false)
+    ColumnTypeEquivalence.alignColumnTypes(mapped10BooleanColumn, Some(true).constant)._2.asInstanceOf[ConstantColumn[Boolean]].value should be(true)
+    ColumnTypeEquivalence.alignColumnTypes(wrappedStringColumn, Some(WrappedString("wrapping paper")).constant)._2.asInstanceOf[ConstantColumn[_]].value should be(WrappedString("wrapping paper"))
+    ColumnTypeEquivalence.alignColumnTypes(trimmedStringColumn, Some("string").constant)._2.asInstanceOf[ConstantColumn[_]].value should be("string")
+    ColumnTypeEquivalence.alignColumnTypes(valueClassStringColumn, Some(ValueClassString("valuable")).constant)._2.asInstanceOf[ConstantColumn[_]].value should be(ValueClassString("valuable"))
+  }
+
+  it should "allow columns whose underlying type is the same to be compared via casting when a global mapped column type is used" in {
+    ColumnTypeEquivalence.alignColumnTypes(castAsString(valueClassStringColumn), "valuable")._2.asInstanceOf[LiteralColumn[_]].value should be("valuable")
+    ColumnTypeEquivalence.alignColumnTypes(castAsString(valueClassStringColumn), "valuable".constant)._2.asInstanceOf[ConstantColumn[_]].value should be("valuable")
+    ColumnTypeEquivalence.alignColumnTypes(castAsOptionString(valueClassStringOptionColumn), "valuable")._2.asInstanceOf[LiteralColumn[_]].value should be(Some("valuable"))
+    ColumnTypeEquivalence.alignColumnTypes(castAsOptionString(valueClassStringOptionColumn), "valuable".constant)._2.asInstanceOf[ConstantColumn[_]].value should be(Some("valuable"))
+    ColumnTypeEquivalence.alignColumnTypes(castAsString(valueClassStringColumn), Some("valuable"))._2.asInstanceOf[LiteralColumn[_]].value should be("valuable")
+    ColumnTypeEquivalence.alignColumnTypes(castAsString(valueClassStringColumn), Some("valuable").constant)._2.asInstanceOf[ConstantColumn[_]].value should be("valuable")
+    ColumnTypeEquivalence.alignColumnTypes(castAsOptionString(valueClassStringOptionColumn), Some("valuable"))._2.asInstanceOf[LiteralColumn[_]].value should be(Some("valuable"))
+    ColumnTypeEquivalence.alignColumnTypes(castAsOptionString(valueClassStringOptionColumn), Some("valuable").constant)._2.asInstanceOf[ConstantColumn[_]].value should be(Some("valuable"))
+    ColumnTypeEquivalence.alignColumnTypes(castAsString(valueClassStringColumn), substring("valuable", 0, 2))._2.columnType should be(StringColumnType)
+    ColumnTypeEquivalence.alignColumnTypes(castAsString(valueClassStringColumn), substring("valuable".constant, 0, 2))._2.columnType should be(StringColumnType)
```

Comment:
Prior to this change the final two assertions here fail when an implicit `String => String` `MappedColumnType` is in scope